### PR TITLE
Add sorting by creation date

### PR DESCRIPTION
**Pull request by Zlopez**
This PR introduces a new version scheme, that is allowing sort by creation_date.
Allows change of version scheme when creating/editing project.
Slight changes how the versions are displayed.

Fixes https://github.com/release-monitoring/anitya/issues/593

### CODE REVIEW

File: anitya/lib/versions/date.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+"""
+
+from .base import Version
+import functools
+
+
+@functools.total_ordering
+class DateVersion(Version):
+    """
+    This class implements an Date version plugin.
```
Comment (bowlofeggs) [inline anitya/lib/versions/date.py]: s/an Date/a Date/

File: anitya/lib/versions/date.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    It sorts versions by date created.
+    """
+
+    name = u'Date'
+
+    def __eq__(self, other):
+        """
+        Compare two versions for equality by date created.
+        """
```
Comment (bowlofeggs) [inline anitya/lib/versions/date.py]: You could add a Returns block since it returns a ```bool```:

```
Returns:
    bool: True if the ```created_on``` attributes are equal.
```

File: anitya/lib/versions/date.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        return self.created_on == other.created_on
+
+    def __lt__(self, other):
+        """
+        Compare two versions by date created.
+        """
```
Comment (bowlofeggs) [inline anitya/lib/versions/date.py]: Similar here since it also returns a ```bool```.

File: anitya/templates/project_new.html [hunk 4]
```diff
@@ -253,6 +262,7 @@ <h1>{{ context }} project</h1>
   }
 
   $(document).ready(function() {
+    add_tooltips();
```
Comment (bowlofeggs) [inline anitya/templates/project_new.html]: This change seems unrelated to sorting by creation date - should it be a separate pull request?

File: anitya/tests/lib/versions/test_date.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2017  Red Hat, Inc.
```
Comment (bowlofeggs) [inline anitya/tests/lib/versions/test_date.py]: s/2017  /2018 /

File: anitya/tests/lib/versions/test_date.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+# should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+import unittest
+
```
Comment (bowlofeggs) [inline anitya/tests/lib/versions/test_date.py]: No empty newline here for PEP-8.

File: anitya/db/migrations/versions/24b6734e8565_creation_date_on_version.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
+"""Creation date on version
+
+Revision ID: 24b6734e8565
+Revises: 34b9bb5fa388
+Create Date: 2018-10-01 11:17:19.457383
+"""
```
[the remark this answers is not in the document]
Comment (Zlopez) [reply anitya/db/migrations/versions/24b6734e8565_creation_date_on_version.py]: This file is generated by alembic. I'm not sure if we should add copyright on this.

File: anitya/lib/versions/date.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+"""
+This adds support for comparing project versions by date created.
+"""
+
+from .base import Version
+import functools
```
[the remark this answers is not in the document]
Comment (Zlopez) [reply anitya/lib/versions/date.py]: I always forgetting about this.

File: anitya/lib/versions/date.py (hunk 2 above)
Comment (Zlopez) [reply anitya/lib/versions/date.py]: You are right.

File: anitya/templates/project_new.html (hunk 4 above)
Comment (Zlopez) [reply anitya/templates/project_new.html]: I did this only for small optimalization, but you are right, I will remove it from this PR.

File: anitya/tests/lib/versions/test_date.py (hunk 6 above)
Comment (Zlopez) [reply anitya/tests/lib/versions/test_date.py]: Strange flake8 didn't threw any error.

**Comment by Zlopez** [comment]
There was a conflict in `anitya.forms` which I resolved when working on the small tweaks.

**Comment by bowlofeggs** [comment]
On Fri, 2018-10-05 at 15:20 +0000, Zlopez wrote:
> Strange flake8 didn't threw any error.

Yeah this is one of the PEP-8 things that I don't think flake8 checks
for. I think there is a checker for this particular thing, but I can't
remember what it was called. I think @beav showed it to me a few years
ago…

**Comment by Zlopez** [comment]
squashed
